[REVIEW] Refactor headers for holtwinters, rproj, tsvd, tsne, umap

chaithyagr opened the pull request:
This is probably the last PR to refactor codes in src/ directory based rules in issue #1675
After this, we will move towards refactoring the primitives.

GPUtester commented:
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

chaithyagr commented:
@teju85 ,

1) I have added to CHANGELOG.md, I wonder how can add it unless I have not opened a PR, so making the process a bit repetitive, please let me know if I missed something.

2) In this PR, I followed the same set of rules. It looks like I found 2 files that included cuda_utils.h, but didnt really call anything from it or had any other device functions, so I removed the include.

3) I found a utils.hxx inside random_projection, which I moved to rproj_utils.cuh, as I think we ave way too many utils.h at this point.

dantegd commented:
@chaithyagr regarding the changelog: indeed you cannot add the line before the PR is created unfortunately. It is not much of a problem for CI (anymore) since CI won't be fully triggered until the commit that adds the changelog entry.

teju85 commented on cpp/src/tsne/utils.h:
```diff
@@ -26,7 +26,6 @@
 #include "linalg/norm.h"
 
 #include <cuda_runtime.h>
-#include <cuda_utils.h>
```
good catch!

teju85 commented:
There seems to be an issue with CI itself here. Tagging @dantegd to make him aware.

teju85 commented:
@chaithyagr can you resolve the conflicts?
